cmd/snap-update-ns/main.go

```diff
@@ -231,22 +232,41 @@ func debugShowChanges(changes []*Change, header string) {
 	}
 }
 
+var osGetenv = os.Getenv
+
 func applyUserFstab(snapName string) error {
 	desiredProfilePath := fmt.Sprintf("%s/snap.%s.user-fstab", dirs.SnapMountPolicyDir, snapName)
 	desired, err := osutil.LoadMountProfile(desiredProfilePath)
 	if err != nil {
 		return fmt.Errorf("cannot load desired user mount profile of snap %q: %s", snapName, err)
 	}
 
+	userHomeSnapDir := ""
+	if ud := osGetenv("SNAP_INSTANCE_USER_DATA"); ud != "" {
+		// /home/joe/snap/foo_bar/123 -> /home/joe/snap
+		if split := strings.SplitN(ud, "/"+snapName+"/", 2); len(split) == 2 {
```

Comment:
What if snap name is `home`.

I think we need something more elaborate than this to be correct.